src/sentry/tasks/derive_code_mappings.py

```diff
@@ -40,29 +40,48 @@ def derive_code_mappings(
     This task is queued at most once per hour per project, based on the ingested events.
     """
     project = Project.objects.get(id=project_id)
-    organization: Organization = Organization.objects.get(id=project.organization_id)
-    set_tag("organization.slug", organization.slug)
+    org: Organization = Organization.objects.get(id=project.organization_id)
+    set_tag("organization.slug", org.slug)
     # When you look at the performance page the user is a default column
-    set_user({"username": organization.slug})
+    set_user({"username": org.slug})
     set_tag("project.slug", project.slug)
-
+    extra = {
+        "organization.slug": org.slug,
+    }
+    feat_key = "organizations:derive-code-mappings"
     # Check the feature flag again to ensure the feature is still enabled.
-    should_continue = features.has(
-        "organizations:derive-code-mappings", organization
-    ) or features.has("organizations:derive-code-mappings-dry-run", organization)
-    if not (dry_run or should_continue):
-        logger.info(f"Event from {organization.slug} org should not be processed.")
+    should_continue = features.has(feat_key, org) or features.has(f"{feat_key}-dry-run", org)
+
+    if not (dry_run or should_continue or data["platform"] not in SUPPORTED_LANGUAGES):
+        logger.info("Event should not be processed.", extra=extra)
         return
 
     stacktrace_paths: List[str] = identify_stacktrace_paths(data)
     if not stacktrace_paths:
         return
 
-    installation, organization_integration = get_installation(organization)
+    installation, organization_integration = get_installation(org)
     if not installation:
         return
 
-    trees: JSONData = installation.get_trees_for_org()
+    trees = {}
+    if not dry_run:
+        trees = installation.get_trees_for_org()
+    else:
+        # Acquire the lock for a maximum of 10 minutes
+        lock = locks.get(
+            key=f"get_trees_for_org:{org.slug}", duration=60 * 10, name="process_pending"
+        )
+
+        try:
+            with lock.acquire():
+                trees = installation.get_trees_for_org(3600 * 3)
```

Comment:
The default caching is 24 hours, however, I want to see improvements quickly. I will revert on the next PR.